docs/how_to/apphooks.rst

```diff
@@ -26,65 +26,78 @@ The file needs to contain a :class:`CMSApp <cms.app_base.CMSApp>` sub-class. For
 
-    class MyApphook(CMSApp):
-        # ...
-        def get_urls(self, page=None, language=None, **kwargs):
-            return [
-                url(r'^$', SomeListView.as_view()),
-                url(r'^(?P<slug>[\w-]+)/?$', SomeDetailView.as_view()),
-                ]
+..  code-block:: python
 
-However, it's neater to keep them in the application's ``urls.py``, where they can easily be reused.
+    url(r'^myapp/', include('myapp.urls', app_name='myapp'))
 
+If you fail to this, then any templates in the application that invoke URLs using the form ``{% url
+'myapp:index' %}`` or views that call (for example) ``reverse('myapp:index')`` will throw a
+``NoReverseMatch`` error.
 
-Apphooks for namespaced applications
-====================================
 
-Does your application use :ref:`namespaced URLs <django:topics-http-defining-url-namespaces>`? This is good practice,
-so it should!
+Apphooks for non-namespaced applications
+----------------------------------------
+
+If you are writing apphooks for third-party applications, you may find one that in fact does
+not have an application namespace for its URLs. Such an application is liable to tun into namespace
+conflicts, and doesn't represent good practice.
 
-In that case you will need to ensure that your apphooks include its URLs in the right namespace. Add an ``app_name``
-attribute to the class that reflects the way you'd include the applications' URLs into your project.
+However if you *do* encounter such a beast, your apphook for it will need to do without the
+``app_name`` attribute.
 
-For example, if your application requires that your project's URLs do::
+Note that unlike apphooks without ``app_name`` attributes can be attached only to one page at a
+time; attempting to apply them a second time will cause an error. Only one instance of these
+apphooks can exist.
 
-    url(r'^myapp/', include('myapp.urls', app_name='myapp')),
+See :ref:`multi_apphook` for more on having multiple apphook instances.
+
+
+Returning apphook URLs manually
+===============================
+
+Instead of defining the URL patterns in another file ``myapp/urls.py``, it also is possible
+to return them manually, for example if you need to override the set provided. An example:
 
-then your ``MyApphook`` class should include::
+..  code-block:: python
 
-    app_name = "myapp"
+    from django.conf.urls import url
+    from myapp.views import SomeListView, SomeDetailView
 
-If you fail to this, then any templates in the application that invoke URLs using the form ``{% url 'myapp:index' %}``
-or views that call (for example) ``reverse('myapp:index')`` will throw a ``NoReverseMatch`` error.
+    class MyApphook(CMSApp):
+        # ...
+        def get_urls(self, page=None, language=None, **kwargs):
+            return [
+                url(r'^$', SomeListView.as_view()),
+                url(r'^(?P<slug>[\w-]+)/?$', SomeDetailView.as_view()),
+                ]
 
```

Comment:
@czpython Before merging, we need to explain how namespacing would be handled when returning URLs manually using this method. What do you suggest?

Reply:
What do you mean? is handled same way